pandas/core/reshape/pivot.py

```diff
@@ -411,9 +411,13 @@ def _all_key(key):
                 table_pieces.append(transformed_piece)
                 margin_keys.append(all_key)
 
-        result = concat(table_pieces, axis=cat_axis)
+        try:
```

Comment:
Table pieces is empty here? Can we just check for
That explicitly instead of try except?

Reply:
done.
I have still returned the df table, as that still contains the cols and index

Reply:
Yep this is ok, just the try except was bothering me